### PR TITLE
add telegram notification module

### PR DESCRIPTION
**Pull request by tyouxa**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

- telegram
##### ANSIBLE VERSION

```
ansible 2.0.1.0
```
##### SUMMARY

 New module for sending notifications via telegram

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @tyouxa for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: notification/telegram.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+- telegram:  token=bot9999999:XXXXXXXXXXXXXXXXXXXXXXX
+    chat_id=000000
+    msg="Ansible task finished"
+
+"""
+
+RETURN = """
+
+msg:
+  description: What message you wish to send
```
Comment (gundalow) [inline notification/telegram.py]: The tense feels wrong here, would it be better as:

"The message you attempted to send"

Comment (tyouxa) [inline notification/telegram.py]: @bob-smith Thanks, I will try to be more readable

File: notification/telegram.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    )
+
+    token = module.params.get('token')
+    chat_id = module.params.get('chat_id')
+    msg = module.params.get('msg')
+
+    if module.check_mode:
+       module.exit_json(changed=False)
+
+    urllib2.urlopen('https://api.telegram.org/'+token+'/sendMessage?text=' + msg + '&'+'chat_id='+chat_id+'').read()
```
Comment (tima) [inline notification/telegram.py]: The variables you are forming the URL with needs to be url quoted for safety.

Comment (tima) [inline notification/telegram.py]: Doesn't catch and handle any exceptions like URLError.

**Comment by gregdek** [comment]
Thanks @tyouxa for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

File: notification/telegram.py [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
 
 
 DOCUMENTATION = """
 
 module: telegram
-version_added: "0.1"
+version_added: "2.1"
```
Comment (tima) [inline notification/telegram.py]: The door has been closed on v2.1 that you'll need to update this to at least 2.2 for Travis to start passing again.

**Comment by tyouxa** [comment]
ready_for_review

File: notification/telegram.py (hunk 3 above)
Comment (tyouxa) [inline notification/telegram.py]: Thanks for help

**Comment by gregdek** [comment]
Thanks @tyouxa for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
Note to self: gundalow_review_this

**Comment by evgkrsk** [comment]
shipit

**Comment by gundalow** [comment]
Just a few minor formatting issues

```
pep8 -r  --ignore=E501,E221,W291,W391,E302,E251,E203,W293,E231,E303,E201,E225,E261,E241,E402 telegram.py 
telegram.py:80:13: E123 closing bracket does not match indentation of opening bracket's line
telegram.py:89:38: E226 missing whitespace around arithmetic operator
telegram.py:89:44: E226 missing whitespace around arithmetic operator
telegram.py:89:65: E226 missing whitespace around arithmetic operator
telegram.py:89:69: E226 missing whitespace around arithmetic operator
telegram.py:89:81: E226 missing whitespace around arithmetic operator
telegram.py:92:8: E111 indentation is not a multiple of four
telegram.py:96:7: E111 indentation is not a multiple of four
telegram.py:98:7: E111 indentation is not a multiple of four

```

needs_revision

File: notification/telegram.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+DOCUMENTATION = """
+
+module: telegram
+version_added: "2.2"
+author: "Artem Feofanov (@tyouxa)"
+
+short_description: module for sending notifications via telegram
+
+description:
+    - Send notifications via telegram bot, to verifyed group or user
```
Comment (gundalow) [inline notification/telegram.py]: `Send notifications via telegram bot, to a verified group or user`

File: notification/telegram.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      module.exit_json(changed=True)
+    else:
+      module.fail_json(msg="failed to send message, return status=%s" % str(info['status']))
+
+
+
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+
```
Comment (gundalow) [inline notification/telegram.py]: As per  http://docs.ansible.com/ansible/developing_modules.html#module-checklist this needs to be:

```
if __name__ == '__main__':
    main()

```

**Comment by gregdek** [comment]
Thanks @tyouxa for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by tyouxa** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @tyouxa for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
Thanks you for promptly addressing the issues raised.

shipit

**Comment by gregdek** [comment]
Thanks again to @tyouxa for this PR, and thanks @ansible/core for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by alikins** [comment]
rebased and squashed this, cleaned up a few tiny style things, and merged to devel 
in https://github.com/ansible/ansible-modules-extras/commit/a8611671ec91c70cdcbbc03cc9c538d8aef05a9e

**Comment by alikins** [comment]
merged to devel in a861167

**Pull request closed without merging**